x-pack/plugins/elastic_assistant/server/lib/langchain/graphs/default_assistant_graph/nodes/run_agent.ts

```diff
@@ -44,11 +46,11 @@ export const runAgent = async ({
     query: '',
   });
 
-  const agentOutcome = await agentRunnable.invoke(
+  const agentOutcome = await agentRunnable.withConfig({ tags: [AGENT_NODE_TAG] }).invoke(
     {
       ...state,
       chat_history: state.messages, // TODO: Message de-dupe with ...state spread
-      knowledge_history: knowledgeHistory?.length ? knowledgeHistory : NO_HISTORY,
+      knowledge_history: JSON.stringify(knowledgeHistory?.length ? knowledgeHistory : NO_HISTORY),
```

Comment:
Without stringify, this will not get parsed when called:
<img width="335" alt="Screenshot 2024-07-01 at 2 03 55 PM" src="https://github.com/elastic/kibana/assets/6935300/4a900a2c-c2ce-42db-a69d-c711ee937dcf">